sse2neon.h

```diff
@@ -3071,6 +3071,14 @@ FORCE_INLINE __m128 _mm_hsub_ps(__m128 _a, __m128 _b)
 #endif
 }
 
+// Horizontally add adjacent pairs of 16-bit integers in a and b, and pack the
+// signed 16-bit results in dst.
+FORCE_INLINE __m64 _mm_hadd_pi16(__m64 a, __m64 b)
+{
+    return vreinterpret_m64_s16(
```

Comment:
For A64, you can utilize `vadd_s16`, `vuzp1_s16` and `vuzp2_s16`. That is, you can implement this intrinsics as following:
```c
    return vadd_s16(vuzp1_s16(a, b), vuzp2_s16(a, b));
```